### ISSUE TITLE
ReferenceError: chrome is not defined

### ISSUE THREAD

**Issue by erosman** [nE0sIghT/chrome-gnome-shell-mirror#114]
I am getting the following error on extensions.gnome.org

Firefox Nightly 103.0a1 (2022-06-19) (64-bit)
Ubuntu 22.04

![image](https://user-images.githubusercontent.com/6458876/174485534-eaa8b125-6f61-4c20-95fc-7d7f80487b10.png)

**Comment by nE0sIghT** [comment]
I can not reproduce this issue in 102.0b9.

Try to disable all other extensions first. Please add console logs. Show me screenshot of "f12 > console > type chrome, type enter"

**Comment by erosman** [comment]
Note: There was no issue before. It has only started recently. 
Here is another screenshot, however `chrome` will not be accessible in Web Developer Tools (F12) since F12 console is DOM console, where chrome/browser (privileged APIs) are not available.

![extensions gnome org](https://user-images.githubusercontent.com/6458876/174491628-8aba1da2-66a9-4c57-9842-18393b284fb2.png)


chrome/browser will be available in Extension Toolbox

![extensions gnome org2](https://user-images.githubusercontent.com/6458876/174492067-173aa369-b24d-4032-85a5-2e2e0fe2079d.png)

**Comment by nE0sIghT** [comment]
Try to report this to Mozilla.
Since you using nightly such issues should not be surprising.

Reopen if it will be reproducible in developer release.

**Comment by erosman** [comment]
Have you tested it with the latest Nightly? 

> Try to report this to Mozilla.

As it is extension related, they will say to contact the extension developer.

**Comment by nE0sIghT** [comment]
Extension works in developer and not works in nightly.
There is no any announcement about `chrome` api deprecation, so this is web-browser bug, not extension bug.

**Comment by erosman** [comment]
While Nightly is subject to many changes before it reaches Beta and release, as an extension developer myself, Nightly gives an early warning about future issues before they hit general public.

I will see what happens with the next Nightly update.

Please also note the error from Browser Console (Ctrl+Shift+J):

```
Error: Please use $(ref:runtime.getURL). 
content-script-start.js:59
```

> [extension.getURL()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extension/getURL)
> 
> Warning: This function is deprecated. Please use [runtime.getURL](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getURL).

**Comment by erosman** [comment]
Errors

> 1655889068252	addons.webextension.chrome-gnome-shell@gnome.org	WARN	Loading extension 'chrome-gnome-shell@gnome.org': Reading manifest: Warning processing externally_connectable: An unexpected property was found in the WebExtension manifest.
>
> 1655889068253	addons.webextension.chrome-gnome-shell@gnome.org	WARN	Loading extension 'chrome-gnome-shell@gnome.org': Event pages are not currently supported. 
> 
>

**Comment by nE0sIghT** [comment]
It is warnings, not errors.  
This is multi-browser extension and this field is valid in Google Chrome and ignored in FF.

**Comment by erosman** [comment]
You are right. It is a warning (I will correct the post). What about "Event pages are not currently supported."?

**Comment by erosman** [comment]
I have been testing. 


```js
// content-script-start.js
console.log(typeof chrome);       // object

// include/sweettooth-api.js
console.log(typeof chrome);       // undefined
```
It could be related to:
```json
  "web_accessible_resources": [
    "include/sweettooth-api.js"
  ],
```

**Comment by Rob--W** [comment]
The extension uses InstallTrigger to detect the Firefox build target: https://github.com/nE0sIghT/chrome-gnome-shell-mirror/blob/6b9334da80714b5bb6202bd17773a9e933db3056/extension/include/constants.js#L15


InstallTrigger has been removed, which stops the browser detection from working.

**Comment by erosman** [comment]
Confirmed .. I set it to `true` and extension works fine now.

```js
COMPAT = {
	IS_FIREFOX:	 true || (typeof(InstallTrigger) !== 'undefined'),
	IS_OPERA:	 navigator.userAgent.indexOf(' OPR/') >= 0
};
```

**Comment by nE0sIghT** [comment]
@Rob--W 

Nice catch! Thanks.  
Since Firefox still lacks `externally_connectable` ([1319168](https://bugzilla.mozilla.org/show_bug.cgi?id=1319168)) we still need a way to detect it and workaround missing feature.

**Comment by erosman** [comment]
For reference only: [Hide InstallTrigger global in Nightly and Early Beta](https://bugzilla.mozilla.org/show_bug.cgi?id=1772905)

**Comment by emilio** [comment]
https://gitlab.gnome.org/GNOME/chrome-gnome-shell/-/merge_requests/8

**Comment by erosman** [comment]
Is there an update for the extension?
I currently have to use an altered version as a temporary addon.

Extension is broken for:

- Firefox 103+ Nightly  released 2022-05-31
- Firefox 103+ Beta released 2022-06-28
- Firefox 103+ due for release on 2022-07-26
